File: src/main/java/com/mrpowergamerbr/loritta/utils/YouTubeUtils.java
package com.mrpowergamerbr.loritta.utils;

import java.util.List;
import java.util.stream.Collectors;

import com.mrpowergamerbr.loritta.Loritta;
import com.mrpowergamerbr.loritta.utils.temmieyoutube.response.SearchResponse;
import com.mrpowergamerbr.loritta.utils.temmieyoutube.utils.YouTubeItem;

public class YouTubeUtils {
	/**
	 * Procura algo no YouTube e retorna o SearchResponse
	 * 
	 * @param query  O que você deseja pesquisar no YouTube
	 * @return O SearchResponse
	 */
	public static SearchResponse searchOnYouTube(String query) {
		SearchResponse response = Loritta.getYoutube().searchOnYouTube(query);
		return response;
	}
	
	public static List<YouTubeItem> searchAnythingOnYouTube(String query) {
		SearchResponse response = searchOnYouTube(query);
		return response.getItems();
	}
	
	public static List<YouTubeItem> searchVideosOnYouTube(String query) {
		List<YouTubeItem> response = searchAnythingOnYouTube(query);
		return response.stream().filter((item) -> item.getId().getKind().equals("youtube#video")).collect(Collectors.toList());
	}
}

File: src/main/java/com/mrpowergamerbr/loritta/utils/misc/YouTubeUtils.kt
package com.mrpowergamerbr.loritta.utils.misc

import com.mrpowergamerbr.loritta.Loritta
import com.mrpowergamerbr.loritta.utils.temmieyoutube.SearchResponse
import com.mrpowergamerbr.loritta.utils.temmieyoutube.YouTubeItem
import kotlin.streams.toList

object YouTubeUtils {
	/**
	 * Procura vídeos no YouTube e retorna o SearchResponse
	 *
	 * @param query  O que você deseja pesquisar no YouTube
	 * *
	 * @return O SearchResponse
	 */
	@JvmStatic
	fun searchOnYouTube(query:String):SearchResponse {
		val response = Loritta.youtube.searchOnYouTube(query)
		return response
	}

	/**
	 * Procura canais, vídeos, livestrams, etc no YouTube e retorna a lista com os resultados
	 *
	 * @param query  O que você deseja pesquisar no YouTube
	 *
	 * @return uma lista com os resultados
	 */
	@JvmStatic
	fun searchAnythingOnYouTube(query:String):List<YouTubeItem> {
		val (_, _, _, _, _, items) = searchOnYouTube(query)
		return items
	}

	/**
	 * Procura vídeos no YouTube e retorna uma lista com os resultados
	 *
	 * @param query  O que você deseja pesquisar no YouTube
	 *
	 * @return uma lista com os resultados
	 */
	@JvmStatic
	fun searchVideosOnYouTube(query: String):List<YouTubeItem> {
		val response = searchAnythingOnYouTube(query)
		return response.stream().filter{ it.id.kind == "youtube#video"}.toList()
	}

	/**
	 * Procura vídeos no YouTube e retorna uma lista com os resultados
	 *
	 * @param query  O que você deseja pesquisar no YouTube
	 *
	 * @return uma lista com os resultados
	 */
	@JvmStatic
	fun searchOnYouTube(query: String, vararg kinds: String): List<YouTubeItem> {
		val response = searchAnythingOnYouTube(query)
		return response.stream().filter{ kinds.contains(it.id.kind)}.toList()
	}
}
